cypress/e2e/swap/errors.test.ts

```diff
@@ -64,10 +62,21 @@ describe('Swap errors', () => {
     })
   })
 
-  it.skip('slippage failure', () => {
-    cy.visit(`/swap?inputCurrency=ETH&outputCurrency=${UNI_MAINNET.address}`, { ethereum: 'hardhat' })
-    cy.hardhat({ automine: false })
-    getBalance(USDC_MAINNET).then((initialBalance) => {
+  it('slippage failure', () => {
+    cy.visit(`/swap?inputCurrency=${USDC_MAINNET.address}&outputCurrency=${DAI.address}`, { ethereum: 'hardhat' })
+    cy.hardhat({ automine: false }).then((hardhat) =>
+      hardhat.fund(hardhat.wallet, CurrencyAmount.fromRawAmount(USDC_MAINNET, 500e6))
+    )
+    cy.hardhat().then((hardhat) => hardhat.mine())
+    cy.hardhat().then(({ approval, wallet }) =>
+      Promise.all([
+        approval.setTokenAllowanceForPermit2({ owner: wallet, token: USDC_MAINNET }),
+        approval.setPermit2Allowance({ owner: wallet, token: USDC_MAINNET }),
+      ])
+    )
+    cy.hardhat().then((hardhat) => hardhat.mine())
```

Comment:
You don't need to keep re-calling hardhat, you can just continue to use promises or chain.
```suggestion
    cy.hardhat({ automine: false }).then(async (hardhat) => {
      await hardhat.fund(hardhat.wallet, CurrencyAmount.fromRawAmount(USDC_MAINNET, 500e6))
      await hardhat.mine()
      await Promise.all([
        approval.setTokenAllowanceForPermit2({ owner: wallet, token: USDC_MAINNET }),
        approval.setPermit2Allowance({ owner: wallet, token: USDC_MAINNET }),
      ])
      await hardhat.mine()
    )
```